resources/user.rb

```diff
@@ -37,11 +38,16 @@ def load_current_value
 end
 
 action :create do
+  package 'openssl' do
+    action :nothing
+  end.run_action(:install)
+
   user new_resource.name do
-    password new_resource.password
+    password generate_system_password
```

Comment:
You can't specify a password there, or it will break https://github.com/sous-chefs/samba/pull/65/files#diff-defe3e036a44c5394d0e7494faad5341R64, as one is clear-text and one is SHA